app/Http/Controllers/API/SuggestController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Http\Controllers\API;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use App\Models\Building;
+use App\Models\Architect;
+
+class SuggestController extends Controller
+{
+
+    public function buildings(Request $request)
+    {
+        $buildings = Building::searchRaw([
+            'suggest' => [
+                'buildings_1' => [
+                    'prefix' => $request->get('search'),
+                    'completion' => [
+                        'field' => 'title.suggest'
+                    ]
+                ],
+                'buildings_2' => [
+                    'prefix' => $request->get('search'),
+                    'completion' => [
+                        'field' => 'title_alternatives.suggest'
+                    ]
+                ]
+            ]
+        ]);
+
+        $ids_1 = array_column($buildings['suggest']['buildings_1'][0]['options'], '_id');
+        $ids_2 = array_column($buildings['suggest']['buildings_2'][0]['options'], '_id');
+        $buildings = Building::findMany(array_merge($ids_1, $ids_2));
```

Comment:
This would save you a trip to the DB:

```suggestion
        $buildings_suggestions = collect(array_merge(
            Arr::get($buildings, 'suggest.buildings_1.0.options'),
            Arr::get($buildings, 'suggest.buildings_2.0.options'),
        ))
        ->pluck('_source')
        ->map(fn ($b) => [
            'id' => Arr::get($b, 'id'),
            'architect_names' => Arr::get($b, 'architect_names'),
            'title' => json_encode([
                'sk' => Arr::get($b, 'sk.title'),
                'en' => Arr::get($b, 'en.title'),
            ])
        ])
        ->toArray();

        $buildings = Building::hydrate($buildings_suggestions)
```

It's more code (and I don't like having to handle the translatable field), but perhaps something to consider...

Reply:
Wow! This is really nice (when I was writing my code - I was thinking, that @eronisko would be able to write it nicer). However I am not 100% sure. I decided to use DB query - because we might want to display also images - which makes query anyway. But right now this is definitely better. So I will take it :). Thanx for effort.